cargo-audit/src/lockfile.rs

```diff
@@ -28,3 +29,141 @@ pub fn generate() -> rustsec::Result<()> {
     }
     Ok(())
 }
+
+// ! Package identifiers.
+// !
+// ! Adapted from Cargo's `package_id_spec.rs`:
+// !
+// ! <https://github.com/rust-lang/cargo/blob/master/src/cargo/core/package_id_spec.rs>
+// !
+// ! Copyright (c) 2014 The Rust Project Developers
+// ! Licensed under the same terms as the `cargo-lock` crate: Apache 2.0 + MIT
```

Comment:
This does not belong here, but I was wondering what structure you would recommend.